instance/tests/models/test_database_server.py

```diff
@@ -333,20 +340,6 @@ def test__create_default(self):
 
         self._assert_default_settings(mysql_server, mongodb_server)
 
-    @override_settings(
-        DEFAULT_INSTANCE_MYSQL_URL=None,
-        DEFAULT_INSTANCE_MONGO_URL=None,
-    )
-    def test__create_default_no_default_settings(self):
-        """
-        Test that `_create_default` does not create database server if default settings missing.
-        """
-        MySQLServer.objects._create_default()
-        MongoDBServer.objects._create_default()
-
-        self.assertEqual(MySQLServer.objects.count(), 0)
-        self.assertEqual(MongoDBServer.objects.count(), 0)
-
```

Comment:
You are right, this is a bit half-baked.  I initially planned to get rid of the option to set the DB servers to `None` altogether, since the default function is already evaluated during the migrations, before you get a chance to manually add an entry.  I then changed the default value function to handle the case that the DB server is unset gracefully, but forgot to add this test back.

However, thinking about it, the current approach is still wrong, since it never errors out, even when it should.  I think I will remove support for not setting default DB servers again, since we eventually want to get rid of ephemeral DB support anyway.  It's the easiest way forward, and I don't think we lose much.